app/models/task.py

```diff
@@ -1,5 +1,50 @@
 from app import db
+from flask import Blueprint, make_response, request, jsonify, abort, request
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, default=None)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'))
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        return {
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at != None,
+            "id": self.task_id
+        }
+
+    @classmethod
+    def from_json(cls, req_body):
+        return cls(
+            title=req_body['title'],
+            description=req_body['description'],
+            #TODO completed_at=req_body['completed_at']
+        )
+
+    def update(self,req_body):
+        try:
+            self.title = req_body["title"]
+            self.description = req_body["description"]
+            self.completed_at = req_body["completed_at"]
+        except KeyError as error:
+            abort(make_response({'message': f"Missing attribute: {error}"}))
+
+    def __str__(self):
+        return "Task"
+
+    def __repr__(self):
+        return "Task"
+        
+    def to_new_dict(self):
+        return {
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at != None,
+            "id": self.task_id,
+            "goal_id": self.goal_id
+        }
```

Comment:
It's kind of odd to make a whole separate `to_new_dict` function here rather than updating `to_dict` to handle adding the `"goal_id"` key. I cloned your code to check if there was another reason, but that was the main use of it in your `GET /tasks/<task_id>` endpoint.

By forcing the call to `to_new_dict` I did see that a test failed because of the `"goal_id"` key. But then that's something that can be handled by saving the dictionary to a variable and then adding the`"goal_id"` key in an `if` block:

```python
    def to_dict(self):
        new_dict = {
            "title": self.title,
            "description": self.description,
            "is_complete": self.completed_at != None,
            "id": self.task_id,
        }

        if self.goal_id:
            new_dict["goal_id"] = self.goal_id
        
        return new_dict
```

Plus, this extra function gets copied over to `Goal`, presumably when you copy-pasted `Task` to create that class. There's nothing wrong with doing that, I do it constantly, but do make sure to check because the `Goal.to_new_dict` function does not ever get used, so there is no need for it.